backend/services/implementations/classService.ts

```diff
@@ -191,6 +193,32 @@ class ClassService implements IClassService {
     }
   }
 
+  async archiveClass(id: string): Promise<ClassResponseDTO> {
+    let archivedClass: Class | null;
+    try {
+      archivedClass = await MgClass.findOneAndUpdate(
+        { _id: id },
+        { isActive: false },
+        {
+          new: true,
+          runValidators: true,
+        },
+      );
+
+      if (!archivedClass) {
+        throw new Error(`Class with id ${id} not found`);
```

Comment:
```suggestion
        throw new Error(`Class with id ${id} not found or not currently active`);
```

Reply:
If it's not currently active, we have to explicitly check for it